example/src/main/java/com/basistheory/android/example/view/dual_write/DualWriteFragment.kt

```diff
@@ -41,22 +41,38 @@ class DualWriteFragment : Fragment() {
         binding.cardNumber.setText("4242424242424242")
         binding.expirationDate.setText("12/25")
         binding.cvc.setText("123")
+
+        binding.cvc.toString() // raw value
+
+        binding.expirationDate.month().toString() // raw value
+
     }
 
+
     private fun createPaymentMethod() = viewModel.client.post(
         "https://api.stripe.com/v1/payment_methods", headers = mapOf(
             "Authorization" to "Bearer {{ STRIPE'S API KEY}}",
             "Content-Type" to "application/x-www-form-urlencoded"
         ), object {
             val type = "card"
+
+            // another option (?) but we don't like it
             val billing_details = object {
-                val name = "Peter Panda"
+                val fullName = object {
+                    val firstName = "Peter"
+                    val lastName = "PANDA"
+                    val expression = "{{ firstName | upcase }} {{ lastName | downcase }}"
+                }
             }
             val card = object {
                 val number = binding.cardNumber
-                val exp_month = binding.expirationDate.month()
-                val exp_year = binding.expirationDate.year()
-                val cvc = binding.cvc
+
+                // these call .toString on ElementReferenceValue and TextElement so we get whatever the default .toString method returns for each
+                // , we want to either return the value without making it available to the outside world
+                // or maybe return some other thing that we can use to grab the element value internally
```

Comment:
> we want to either return the value without making it available to the outside world

If we make `toString` internal kotlin won't like it bc we're weakening visibility of the function and it is different from the interface and it shouldn't work anyways because this is called in the client app 

> or maybe return some other thing that we can use to grab the element value internally

I've not sure if there's pattern to get the element instance from the HTTP client when calling `getElementsValues`